### PR TITLE
Added --ignore-missing option to cope with missing bcl files

### PR DESCRIPTION
**Pull request by jenniferliddle**

### CODE REVIEW

File: src/i2b.c [hunk 1]
```diff
@@ -1522,7 +1528,7 @@ static void *bcl_thread(void *arg)
     }
     if (!bcl) {
         bcl = openBclFile(o->opts->basecalls_dir, o->lane, o->tile, o->cycle, o->surface, o->tileIndex, o->filter);
-        if (bcl->errmsg) { display("%s", bcl->errmsg); bcl = NULL; }
+        if (bcl->errmsg) { display("%s", bcl->errmsg); /* JSL bcl = NULL; */ }
         if (o->bcl_cache) {
             if (bcl) insert_bclfile_to_cache(bcl, o->bcl_cache, o->lane, o->cycle, o->surface);
```
Comment (srl147) [inline src/i2b.c:1533]: I assume this wasn't changed because you still want to add un-opened files to the cache

Comment (jenniferliddle) [reply src/i2b.c:1533]: Exactly so.

**Comment by srl147** [comment]
the new test data novaseq_1_missing.sam appears to be a bam file

**Comment by jenniferliddle** [comment]
> the new test data novaseq_1_missing.sam appears to be a bam file

Yes. That's because of , errr, historical reasons.

**Comment by srl147** [comment]
I should I only noticed because I tried to read it, the test obviously doesn't care
________________________________________

**Pull request merged**